### PR TITLE
Feat/optional auth

### PR DESCRIPTION
**Pull request by mlsandar**
I'll generate a comprehensive PR description for your optional authentication feature. Based on all the work we've done, here's a complete description:

---

# 🔐 Add Optional Authentication Support for Jobmon Server and GUI

## Overview

This PR implements optional authentication for both the Jobmon server and GUI, allowing the system to operate without authentication requirements for development, testing, and demonstration environments. When disabled, the system provides a seamless anonymous user experience while maintaining all existing functionality.

### Server-Side (Phase 1)
- ✅ **Configurable Authentication**: Control via `JOBMON__AUTH__ENABLED` environment variable
- ✅ **Anonymous User Support**: Automatic anonymous user creation when auth is disabled
- ✅ **Conditional OAuth Setup**: OAuth configuration only when authentication is enabled
- ✅ **API Dependency Management**: Conditional user validation based on auth status
- ✅ **CORS Configuration**: Dynamic CORS setup based on authentication requirements

### Frontend (Phase 2)  
- ✅ **Environment-Based Control**: Toggle via `VITE_APP_AUTH_ENABLED` environment variable
- ✅ **Anonymous User Experience**: Seamless operation without login screens
- ✅ **Conditional UI Elements**: User menu and auth indicators based on auth status
- ✅ **Axios Configuration**: Conditional credential sending based on auth status
- ✅ **Visual Feedback**: Clear indicators when authentication is disabled

### Backend Changes
- **`jobmon_server/src/jobmon/server/web/routes/utils.py`**: Added auth configuration checks and anonymous user functions
- **`jobmon_server/src/jobmon/server/web/api.py`**: Updated API router dependencies and CORS configuration
- **`jobmon_server/src/jobmon/server/web/auth.py`**: Made OAuth setup conditional

### Frontend Changes
- **`jobmon_gui/src/contexts/AuthContext.tsx`**: Added environment-based auth toggle with anonymous user support
- **`jobmon_gui/src/components/navigation/PageNavigation.tsx`**: Conditional user menu and auth status indicator
- **`jobmon_gui/src/configs/Axios.ts`**: Conditional credential configuration

### CORS Fix
- **Resolved XMLHttpRequest CORS policy error** when credentials mode is 'include'
- **Conditional CORS origins**: Wildcard for no-auth, specific origins for auth mode
- **Configurable origins**: Support for `JOBMON__CORS__ALLOWED_ORIGINS` configuration

### To Disable Authentication:

**Server-side:**
```bash
JOBMON__AUTH__ENABLED=false
```

**Client-side:**
```bash
VITE_APP_AUTH_ENABLED=false
```

### Default Behavior
- Authentication is **enabled by default** (backwards compatible)
- Both server and client settings must be `false` to fully disable authentication
- Production environments should keep authentication enabled

## 🧪 Testing

- ✅ **16 comprehensive server tests** covering all authentication scenarios
- ✅ **6 frontend tests** for AuthContext component behavior
- ✅ **TypeScript compilation** without errors
- ✅ **Build verification** for both server and frontend
- ✅ **CORS validation** in both auth modes

### Test Coverage
- Configuration edge cases (undefined, empty values)
- Anonymous user creation and consistency
- API dependency injection based on auth status
- Frontend conditional rendering
- Environment variable integration

## 📚 Documentation

- ✅ **README.md**: Added developer setup guide for optional authentication
- ✅ **CHANGELOG.md**: Documented feature in main project changelog
- ✅ **jobmon_gui/CHANGELOG.md**: Documented GUI changes and usage plot refactor

## 🎯 Use Cases

- **Local Development**: Run without OAuth setup requirements
- **Testing Environments**: Simplified testing without authentication overhead
- **Demonstrations**: Easy showcase of functionality without login barriers
- **CI/CD Pipelines**: Automated testing without authentication complexity

## 🔄 Backwards Compatibility

- ✅ **No breaking changes**: Authentication enabled by default
- ✅ **Existing workflows**: Continue to work without modification
- ✅ **Production ready**: No impact on production deployments

### For Development Environments:
1. Add `JOBMON__AUTH__ENABLED=false` to your server `.env` file
2. Add `VITE_APP_AUTH_ENABLED=false` to your `jobmon_gui/.env` file
3. Restart both server and frontend
4. Access the application without authentication

### For Production:
- No changes required - authentication remains enabled by default

## 📸 Visual Changes

When authentication is disabled:
- Login screen is bypassed
- User menu is hidden in navigation
- "🔓 Authentication Disabled" indicator appears in header
- All functionality works with anonymous user context

## 🔗 Related

- Fixes CORS issues when switching between auth modes
- Enables easier development and testing workflows

---

**Testing Instructions:**
1. Set environment variables as described above
2. Start server and frontend
3. Verify no login screen appears
4. Confirm all API calls work without authentication
5. Check that auth indicator appears in navigation

This PR provides a foundation for easier development while maintaining production security standards.

### CODE REVIEW

File: jobmon_gui/src/components/navigation/PageNavigation.tsx [hunk 1]
```diff
@@ -110,6 +110,9 @@ export default function PageNavigation({ children }: PropsWithChildren) {
     const [userMenuAnchorEl, setUserMenuAnchorEl] =
         useState<null | HTMLElement>(null);
 
+    // Check if auth is enabled
+    const authEnabled = import.meta.env.VITE_APP_AUTH_ENABLED !== 'false';
```
Comment (everdyke) [inline jobmon_gui/src/components/navigation/PageNavigation.tsx:114]: === 'true' would be a stricter check.
